sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/CBDE/CheckMLIRWithReference.cs

```diff
@@ -0,0 +1,176 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace SonarAnalyzer.UnitTest.CBDE
+{
+    [TestClass]
+    public class CheckMlirWithReference
```

Comment:
should you also add tests for
- Bitwise AND, OR, XOR operations?
- multiplication, division , modulo?

Reply:
I don't really think so, because those operations are handled in the same way as operators that are tested here, and in the same functions (`ExtractBinaryExpression` & `ExtractBinaryAssignmentExpression`), so adding tests here would not increase coverage.

Reply:
I'm not talking about increasing coverage (_pleasing the tool_), I am talking about testing the behavior of our analyzers.

I believe tests should not be made based on implementation, but based on functionality. If the implementation is refactored (e.g. a `switch` will be replaced with `if-else` branches), tests should stay the same.

Reply:
tests were added